app/[lang]/chapters/[slug]/[lesson]/page.tsx

```diff
@@ -23,5 +73,47 @@ export default function Page({ params }) {
 
   const Lesson = chapterLessons[params.lesson].default
 
+  if (dev) {
+    return <Lesson lang={params.lang} />
+  }
+
+  // If account or progress data is being loaded, we show a loader.
+  if (unlocked === LoadingState.Idle || isAccountLoading || isProgressLoading) {
+    return (
+      <div className="flex grow items-center justify-center">
+        <Loader className="h-10 w-10 text-white" />
+      </div>
+    )
+  }
+
+  // If account and progress data have been loaded, but there is no progress obj
+  if (!account || !progress) {
+    return (
+      <div className="flex h-full w-full grow flex-col items-center justify-center">
+        <span className="mb-10 font-nunito text-2xl text-white">
+          Sorry, you need to be signed in to access this lesson
+        </span>
+        <Button onClick={handleSignInClick} size="small">
+          Sign in
+        </Button>
+      </div>
```

Comment:
Assuming the user has to sign in to proceed can we just render the `Modal.SignIn` here to save a click instead of prompting them?